### DISCUSSION TITLE
How to edit file in the same window as a new buffer in lazygit?

### DISCUSSION THREAD

**Discussion by kcaashish** [LazyVim/LazyVim#543] [Q&A]
I cannot figure this out. Changed `config.yml` for lazygit to use `nvim`, used `neovim-remote`, changed `.zshrc` as suggested but I cannot figure out how to open/edit file from lazygit as a new buffer in the same window, and not in the floating terminal.

`config.yml`:
```yml
os:
  editCommand: "nvim"
  editCommandTemplate: "{{editor}} +{{line}} -- {{filename}}"
```

to `.zshrc`:
```sh
if [[ -n $NVIM_LISTEN_ADDRESS ]]; then
  alias nvim="nvr -cc split --remote-wait +'set bufhidden=delete'"
fi

if [[ -n $NVIM_LISTEN_ADDRESS ]]; then
  export VISUAL="nvr -cc split --remote-wait +'set bufhidden=delete'"
  export EDITOR="nvr -cc split --remote-wait +'set bufhidden=delete'"
else
  export VISUAL="nvim"
  export EDITOR="nvim"
fi
```

Even added this to `config/lazy.lua`:
```lua
if vim.fn.executable("nvr") == 1 then
  vim.env.GIT_EDITOR = "nvr -cc split --remote-wait +'set bufhidden=delete'"
end
```

Still opens in the floating terminal.

**Comment by mapringg** [comment]
I got it working without using neovim-remote, but I am on the latest HEAD.

This is what I added to my lazygit config:

```yml
os:
  editCommand: "nvim"
  editCommandTemplate: "{{editor}} --server $NVIM --remote-tab {{filename}}"
```

**Reply by kcaashish** [reply]
When I use this, the file looks like it got opened, the name of the file is displayed as it is, but it opens as an empty buffer. When I open the file using telescope, it directly opens in Insert mode. So, it looks like the edit functionality is working but I cannot get this to work properly. Did you do anything else?

**Reply by mapringg** [reply]
I don't remember doing anything else beside installing the latest HEAD. This is my output after I exit out of editing if my lazygit is run outside neovim: 

<img width="572" alt="Screenshot 2566-04-10 at 22 59 56" src="https://user-images.githubusercontent.com/15067237/230941179-471e2c83-bf06-4c37-8a2e-a205a6c1fe76.png">

**Reply by kcaashish** [reply]
Okay, if the nvim is opened for the very first time this way, if I open the lazygit outside then do this, it seems to work.

**Reply by smilovanovic** [reply]
@mapringg works perfectly, thanks!

**Reply by folke** [reply]
That's really useful. Didn't know that worked. Thanks for sharing! :)

**Reply by krishna-bala** [reply]
This works for me, but unfortunately does not work with `C` (`commit changes using git editor`). I have my lazygit `config.yml` configured like so:

`config.yml`
```yaml
os:
  editCommand: "nvim"
  editCommandTemplate: "{{editor}} --server $NVIM --remote-tab {{filename}}"
```

Using the `config.yml` above, when I open lazygit from inside neovim (as a floating window) and open/edit files (from the lazygit floating window), it opens the file in the original neovim instance (not in the floating window). But when I try to "Commit changes using git editor", it opens a nested instance of neovim inside of the floating window.

I see an [open issue](https://github.com/jesseduffield/lazygit/issues/2045) on the lazygit repo that mentions this issue, but it looks like the [PR](https://github.com/jesseduffield/lazygit/pull/2046) was never merged. However, on the PR, the lazygit creator suggested 
> What are your thoughts on supplying core.editor to git in the git commit command e.g. as described here: https://stackoverflow.com/questions/15884180/how-do-i-override-git-configuration-options-by-command-line-parameters

I'm not sure what to do given that suggestion, so I thought I would reach out here first in case anyone has a solution. I'm happy to post this on the lazygit repo issue/PR too.

**Reply by kcaashish** [reply]
I have not managed to get this to work when I open nvim with the command `nvim`, then use lazygit for the first time inside nvim, it just opens a blank file. However if I open lg in the terminal and edit from there, it works inside nvim for rest of the steps. I will post a video of my problem when I get the chance.

**Reply by kcaashish** [reply]
[Screencast from 2023-05-04 13-13-58.webm](https://user-images.githubusercontent.com/31175326/236138408-83cf70cb-25be-4d26-a35d-a752539d9de5.webm)

**Reply by kcaashish** [reply]
I think it was an issue with alacritty or Fedora, I tried in in iTerm2 and it works. The new recommended way from `lazygit` following this answer:
```yml
os:
  edit: "nvim --server $NVIM --remote-tab {{filename}}"
```

**Reply by iamxiaojianzheng** [reply]
The environment I use is Windows terminal + zsh + nvim.

I have the same situation with you, when I open lazygit in lazyvim and then open git file option through e, nvim is stuck directly.

When I opened lazygit directly in zsh via lg and then opened git file option via e, I was able to edit it smoothly in nvim.
